File: GXpert/GXpert.Web/Modules/Institute/InstituteTimeTable/InstituteTimeTableColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Institute.Columns;

[ColumnsScript("Institute.InstituteTimeTable")]
[BasedOnRow(typeof(InstituteTimeTableRow), CheckNames = true)]
public class InstituteTimeTableColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public DateTime Date { get; set; }
    public DateTime StartTime { get; set; }
    public DateTime EndTime { get; set; }
    public int PeriodIndex { get; set; }
    public string TeacherPrn { get; set; }
    [QuickFilter]
    public string InstituteName { get; set; }
    [QuickFilter]
    public string DepartmentName { get; set; }
    [QuickFilter]
    public string BranchName { get; set; }
    [QuickFilter]
    public string InstitudeDivisionName { get; set; }
    public short EType { get; set; }
    public int ClassRoomNo { get;set; }
    public short TeacherAttendanceStatus { get; set;}
    
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Institute/InstituteTimeTableColumns.ts
﻿import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { EInstituteTimeTableType } from "../Web/Enums.EInstituteTimeTableType";
import { InstituteTimeTableRow } from "./InstituteTimeTableRow";

export interface InstituteTimeTableColumns {
    Id: Column<InstituteTimeTableRow>;
    Date: Column<InstituteTimeTableRow>;
    StartTime: Column<InstituteTimeTableRow>;
    EndTime: Column<InstituteTimeTableRow>;
    PeriodIndex: Column<InstituteTimeTableRow>;
    TeacherPrn: Column<InstituteTimeTableRow>;
    InstituteName: Column<InstituteTimeTableRow>;
    DepartmentName: Column<InstituteTimeTableRow>;
    BranchName: Column<InstituteTimeTableRow>;
    InstitudeDivisionName: Column<InstituteTimeTableRow>;
    EType: Column<InstituteTimeTableRow>;
    ClassRoomNo: Column<InstituteTimeTableRow>;
    TeacherAttendanceStatus: Column<InstituteTimeTableRow>;
}

export class InstituteTimeTableColumns extends ColumnsBase<InstituteTimeTableRow> {
    static readonly columnsKey = 'Institute.InstituteTimeTable';
    static readonly Fields = fieldsProxy<InstituteTimeTableColumns>();
}

[EInstituteTimeTableType]; // referenced types